### PR TITLE
Locks up Medical Storage a bit more tightly and adds framework for doors to ignore access reqs during red alert

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
balance: Medical supplies on all normal maps (Box, Meta, Delta, Omega, and Pubby) are now behind ID-locked windoors. These windoors will become all-access during red alert!
/:cl:

Having played a bit more medical lately, and talking to a lot of people who do so more actively than me, I've discovered that ~~polina~~ people helping themselves to the medical supplies is commonplace, and because Medical is treated like an open department by most of Security who are more concerned with getting valids, they don't usually care about such theft, which makes it very commonplace.

To fix this a little, I added windows around the first aid kit storage on all the maps, with windoors that anyone with Medbay access can open. You can see the placement below. This makes it harder to waltz in and take literally all the medkits, but still gives doctors easy access and allows stealing them after a few seconds more instead of instantly.

These windoors also unlock automatically in the event of a red alert, so people can access crucial medical supplies without having to yell at or loot someone else for their ID. Code-wise, this is done with a new `red_alert_access` variable that overrides the `check_access()` and `check_access_list()` procs to return `TRUE` if the security level is at least red alert.

I tried to make sure the windows and their doors weren't obtrusive by overlaying them on top of existing tables.

Alternative to #33695.

Boxstation
Metastation
Deltastation
Omegastation
Pubbystation

### CODE REVIEW

**Comment by awesine** [comment]
"Having played a bit more medical lately, and talking to a lot of people who do so more actively than me, I've discovered that polina people helping themselves to the medical supplies is commonplace, "
this is a good thing

**Comment by Xhuis** [comment]
Medical is its own department, not public space. In a place that only doctors can access, All other rooms in the department are treated like a restricted workspace due to the ID restrictions; if you'll be thrown out of surgery, chemistry, genetics, and virology for breaking in. the most important part of medbay (where their supplies are, since chemistry usually doesn't bother) is logically restricted, too. Otherwise the doors would just be open to everyone!

**Comment by awesine** [comment]
@Xhuis 
yeah you'll get thrown out but then that shitter will just stun/cuff you and take what he wanted anyway or he's going to get his buddies and beat up on you

both of these are valid because you probably need healing and there's not much you can do to avoid getting brainfucked rather than just giving them medkits and not making the entire station hate you

**Comment by epochayur** [comment]
It's really not. Having assistants sprint in, take an entire kit for a single ointment/brute pack, stashing the whole thing in their bag then running off and dying massively fucks with medbay later in the round when the supply room gets cleaned out in the first 20 minutes. If you want healing, talk to a doctor. If there's no doctors, get cargo to order a crate. If both of those fail, then just break in like normal. This is all considering that the station isn't already on red alert, which it is a lot of the time, in which case you can grab a medkit anyway. 

Great PR

**Comment by Xhuis** [comment]
Why are you trying to defend clearly tidey and assholeish behavior from that kind of player, awesine? Being a dick to other players shouldn't be a valid excuse for breaking in and stealing things. If you need healing, there are sleepers and cryo that *are* publically available. Stealing medkits exists only to serve yourself.

**Comment by awesine** [comment]
@Xhuis 
you can't self-service cyro and sleepers

**Comment by Xhuis** [comment]
@Cyberboss WHAT DID YOU DOOOO

**Comment by awesine** [comment]
@epochayur 
there are two massively easier options for medical that are better than medkits
spend .5 seconds making a beaker in chemistry and feed that beaker to your patient
spend .5 seconds throwing him into a sleeper and pressing buttons
not spending .5 seconds taking medkits that other people like security need much more than you fuckin do

**Comment by epochayur** [comment]
Alright, do that then. I fail to see what you're complaining about. 
>not spending .5 seconds taking medkits that other people like security need much more than you fuckin do
What point are you even trying to make here?

**Comment by awesine** [comment]
the point is that other people that WILL get in combat and WON'T stay in their autism fortress won't always have the time to wait 5 minutes for medbay to realize you're actually in pain or actually have the time to get to medbay

either way, this won't stop anything, this is just going to cause more station v medbay conflict and make players have to break more shit to get what they need/want

**Comment by coiax** [comment]
I like this PR, even if windoors are a complete pain to repair/replace.

But I think you need help massaging the commits into the right place.

**Comment by KorPhaeron** [comment]
Medical players are so weird

Why do we need fort knox for medkits when medbay is full of infinite chem machines

**Comment by awesine** [comment]
above

**Comment by epochayur** [comment]
>the point is that other people that WILL get in combat and WON'T stay in their autism fortress won't always have the time to wait 5 minutes for medbay to realize you're actually in pain or actually have the time to get to medbay

Yeah, Alex, I'll take "things that never happen" for 500. In the rare event that
1) No doctors, CMOs, geneticists, chemists, sec officers, medibots, medborgs or virologists are present in medbay
2) The chem fridge is not at all stocked
3) Cargo hasn't ordered crates
4) The numerous medkits strewn about the station have been taken
5) The supply room doors aren't on emergency access
6) The station somehow still isn't on red alert at this point

Then sure, you might have to break in, which with the glass around the medpacks, will take an extra 5 seconds to screwdrive open. What's the issue here?

**Comment by awesine** [comment]
>""things that never happen""
i can recall 12 times this has happened to me in the past 3 days

**Comment by awesine** [comment]
i can recall MORE times when there were absolutely no doctors in sight and breaking into med storage was my only chance of not bleeding out

**Comment by epochayur** [comment]
>Then sure, you might have to break in, which with the glass around the medpacks, will take an extra 5 seconds to screwdrive open.

**Comment by Xhuis** [comment]
@KorPhaeron Imo, because the chems *are* infinite and are limited to medbay, whereas the main supplies for any medical staff trying to treat someone not at medbay itself are limited and easy to steal.

**Comment by awesine** [comment]
@epochayur 
you have to get someone to open the windoor to actually screwdriver it open
another example of coderbus never actually playing

**Comment by awesine** [comment]
and what if you can't afford to climb over the glass or you die

**Comment by epochayur** [comment]
there's literally window panes right there that take exactly five seconds to screwdriver -> crowbar -> screwdriver

**Comment by Xhuis** [comment]
if you break into the storage, then you already have all the tools necessary to get to the supplies, my dude; either decon the window (you need a screwdriver and crowbar to hack a door, meaning you can do this) or smash it (you need to smash the window to get in without the door, and the supplies are blocked by other windows)

every medical kit can be accessed without climbing on any table; the windoors were placed in this way and you can smash the windows there too

**Comment by awesine** [comment]
still extra time you have to use to get the medkits when they could just be public access and you don't gotta wait an hour for some bored ass medical person to come to medbay/realize you're hurt

and there's a better alternative to medkits
RIGHT NEXT TO THE MEDKITS
*it's called the med dispenser*

**Comment by Xhuis** [comment]
medbay is not designed to be public space; no department is *designed* to be self-service. you can't do research without supplies and manpower, you can't make chemicals, viruses, or genetic mutations without people doing them

it not designed to be self-service because the medkits are *behind ID-locked doors* and smashing the entrance does not change reality just because you made a new door

**Comment by epochayur** [comment]
>when they could just be public access

see
>Seems like a small tradeoff to keep supply being ransacked early and medbay being woefully under-supplied later in the round.

I wouldn't be against having one or two extra medkits and maybe a toxin kit sitting in medbay for people to grab, but my point remains.

**Comment by awesine** [comment]
medbay is never woefully undersupplied
never
they are never
never
under any circumstance
under-supplied
the only
**only**
chance medbay would be undersupplied
if it was bombed
why are they never woefully undersupplied?
They have  med dispenses
they have sleepers
they have chem dispensers

**Comment by epochayur** [comment]
You can say that all you like, but medbay gets cleaned out more often than not

**Comment by awesine** [comment]
i'm not convinced you actually read what i posted

try again

**Comment by epochayur** [comment]
>You can say that all you like, but medbay gets cleaned out more often than not

**Comment by awesine** [comment]
>i'm not convinced you actually read what i posted

try again

**Comment by Xhuis** [comment]
Here's my own two cents:

the medkits stored away are most useful for healing large amounts of damage; this is due to the bruise packs, ointment, and patches all having instant application to heal big chunks of it at once

sleepers and cryo heal smaller wounds but are not useful on people with extreme or life-threatening damage; this is evidenced by sleepers not working when someone has high damage unless it's upgraded, and cryo being quite slow since the primary medicine is mainly used to treat clone damage

nanomeds have only a very small amount of medicines, and in the most recent rounds of mine, chemistry never produces enough to satiate the needs of a busy medbay in a full-swing highpop round or emergency where multiple dead and dying come in rather quickly

people take medkits to heal any amount of damage on them, meaning that bruise packs and ointment are often wasted treating minor wounds because people take the supplies for themselves, and if a lot of severely-wounded or dying patients come in, there are often not enough supplies to go around because people take them for themselves to treat their wounds and then leave with them

example A: a traitor goes on a rampage, killing four people and highly damaging another; the damaged person goes into medbay, takes a first aid kit, heals himself, and stashes the supplies for later. the AI, as well as other medical staff, could have put them into a sleeper or cryo to stabilize them since the wound wasn't killing them, but they took the supplies instead

chaos like this, coupled with all the minor wounds and injuries rounds accrue, mean that the white first aid kits go very quickly - there are only four!! - leaving doctors with the supplies they start with and the specialized first aid kits which are currently not as useful as the generalized ones

**Comment by awesine** [comment]
@Xhuis 
>sleepers and cryo heal smaller wounds but are not useful on people with extreme or life-threatening damage

you say this but the game says otherwise

**Comment by Xhuis** [comment]
The majority of the last rounds I've played this month have been as medbay staff. sleepers literally will not allow you to inject any reagent other than epinephrine if the person is in critical condition, and cryo heals far more slowly than other medicines - if you put someone ODing on bicard in there then cryoxadone alone is not enough to outpace the already-slow brute damage they're taking

**Comment by awesine** [comment]
that's why you go into chemistry and spend .5 seconds making a healing chem and forcefeed it to the patient

**Comment by KorPhaeron** [comment]
>cryo heal smaller wounds but are not useful on people with extreme or life-threatening damage

what

**Comment by awesine** [comment]
cyro is literally meant for people in critical condition

that is the only reason cyro exists

coders do not play the game

**Comment by Xhuis** [comment]
chemistry is *locked*; medical doctors aren't allowed to waltz in for the same reason that people can't waltz into storage, and I guarantee chemists are a lot more stingy about it. if chemistry provides infinite healing supplies, then why do the first aid kits exist at all instead of just chem dispensers with the recipes for styptic powder and silver sulf. taped on the front?
@KorPhaeron I provided my reasoning for that statement directly following it, of course it sounds stupid with no context
@awesine ok bud

**Comment by QualityVan** [comment]
I mean, this won't fix self-serve medical existing and being cancerous as shit, but it's not a bad change.

ps fold chemistry into meddoc 2018

**Comment by awesine** [comment]
@Xhuis 
first aid kits so that people in the field can heal themselves without being transferred to a hospital
the real, absolute 100% world-life reason why med-kits exist

**Comment by Xhuis** [comment]
then why are there r-walls in the spaces where storage intercepts maintenance or other public areas, and why is it behind two ID-locked doors (medbay entrance and the airlocks itself)? if people can just take them on their own, then why not add a public door to there? also, you still haven't answered my question about the chem dispensers

**Comment by awesine** [comment]
@Xhuis 
because mappers are retarded

>if chemistry provides infinite healing supplies, then why do the first aid kits exist at all instead of just chem dispensers with the recipes for styptic powder and silver sulf. taped on the front?
chemistry's job is to make healing chems for medical personnel

**Comment by Xhuis** [comment]
then there should be no first aid kits to begin with, if chemistry already produces infinite healing; i mean hey, you can just make some patches and stuff 'em in your backpack for when you need them in the field

**Comment by awesine** [comment]
@Xhuis 
first aid kits are spawned so that combat personnel/people who need them can take them and use them when they get hurt

how the actual fuck are you not getting this

you are hurting my brain with how dumb you are being

**Comment by awesine** [comment]
i'm opening up an alternative to this within 2 minutes anyway

**Comment by Xhuis** [comment]
you seem to be intent on pushing your own argument and not attempting to understand the reasons for mine, so this entire conversation is a moot point to begin with, but good luck with the alternative

**Comment by Anonmare** [comment]
I already lock up the medkits anyway because assistants deserve nothing but inconvenience so this just saves me time

**Comment by awesine** [comment]
you seem to be intent on pushing your own argument and not attempting to understand the reasons for mine, so this entire conversation is a moot point to begin with, but good luck with the alternative

**Comment by Xhuis** [comment]
you have a pleasant day! thanks for discussing it anyway, it does help me see other views on it better, though I'm still curious about kor's own reasoning. y'all are both entitled to your own thoughts, but I'd request that you, specifically, not call me "fucking stupid" for having my own

**Comment by awesine** [comment]
https://i.imgur.com/nx5kEzO.png

**Comment by awesine** [comment]
you also might wanna fix the commits for this

**Comment by Xhuis** [comment]
that's apparently something @Cyberboss did; it seems to be on all PRs opened about 5-10 minutes after this one was, but the details are a bit hazy right now so I'm waiting for word back before I do anything

**Comment by awesine** [comment]
did you just alternative an alternative

**Comment by Xhuis** [comment]
I figured it'd help clarity to state that it was the opposite PR to yours.

**Comment by Xhuis** [comment]
Alright, I did some verification, and I was completely wrong about cryo. It's a lot faster than I thought it was and it definitely works as more than just a way to fix minor wounds. I'm sorry for fighting for that; it was dumb of me. I still stand by this PR, but on that front, I was completely wrong.

**Comment by jughu** [comment]
Medsleepers only matter when science upgrades them

File: _maps/map_files/Deltastation/DeltaStation2.dmm [hunk 1]
```diff
@@ -73304,7 +73310,6 @@
 /area/medical/storage)
 "cSV" = (
 /obj/machinery/atmospherics/pipe/simple/supply/hidden,
-/obj/machinery/rnd/protolathe/department/medical,
```
Comment (ShizCalev) [inline _maps/map_files/Deltastation/DeltaStation2.dmm]: Deleted the lathe?

**Review by ShizCalev — DISMISSED** [review]

**Comment by awesine** [comment]
Lock up the sleepers too so non-medbay personnel can't use them! And cyro!

**Comment by epochayur** [comment]
https://i.imgur.com/mG8AuDC.jpg

**Comment by ShizCalev** [comment]
Keep in mind this'll also affect nuke ops, malf, cult, & blob.

**Comment by Xhuis** [comment]
Yup. I don't want the supplies to be locked down incredibly tight, since they're supplies and people will need to use them - just want to secure them enough that people can't just break in and help themselves.

**Comment by deathride58** [comment]
The thing about medkits is that ointment and bruise packs are actually much more effective at healing than most patches/pills. In most cases, one standard bruise pack/ointment is enough to get up to !!!6!!! people out of crit, at which point they can safely be shoved into a sleeper to inject bicardine or kelotane. With a standard medkit having two bruise packs, and two ointment stacks, that means a standard medkit can heal up to !!!24!!! different traitor victims out of crit.

Assistants stealing medkits for tiny scratches should be shot out of a cannon, as medkits aren't intended for minor scratches in any way, shape, or form. Sleepers and chem patches/pills are what assistants should be after if they have a scratch.

This PR is a very, very good change because of how useful medkits are in most cases.

~~Merging chemist and medical doctor together would also be a pretty great change, as well, but that's for a different discussion~~

**Comment by AdamElTablawy** [comment]
Good pr.

On the grounds that medkits are for whatever IRL, it doesn't fucking matter because this is a video game. There are doctors in medbay. You go to medbay to heal. Self-service is possible, but it is intentionally made less available than being able to be healed in medbay. A design that encourages this is a good thing.

**Comment by KorPhaeron** [comment]
Over reinforcing things is dull, one of the most common mistakes mappers make is that we are not trying to design a functional space station because this isn't real life we're trying to design a place that is fun to have disasters in

Things being stolen is a natural and intentional part of gameplay (antags need to be able to do it for obvious reasons, and a certain number of non antags need to be doing it to give security something to do) and medical supplies are not rare and valuable enough to warrant this level of protection

**Comment by Xhuis** [comment]
The majority of insulated gloves are in ID-locked closets, space suits are behind multiple airlocks and often shocked windows in high-security areas, and spare gas is very hard to get, yet all of these things are common. 

Imo, the first aid kits are valuable for the reasons people already stated about the crazy efficiency of them (mainly the white ones), and having them stolen often affects them more than the objects above because chemistry is unreliable the majority of the time, and one job (MD) having to do another job (chemist) just to have the supplies to do help people is bad design in my eyes.

**Review by Cyberboss — APPROVED** [review]

**Comment by cacogen** [comment]
if the doctors are doing their jobs nobody needs to steal medical supplies. plus you've just added one more window to break.

**Comment by Iamgoofball** [comment]
@Xhuis 
I've tried to fight "medbay is not a public space" many times, it's not worth it

**Comment by PaznerIV** [comment]
As someone who played med bay quite often I will say this.  If you don't give an instetive to seek a medical doctor in the first place then players will self sever, but if you give a positive feed back loop for seeking out doctors then players would be more willing to try to reach out for the medical staff. Just adding more obstacles to reach the med kits will deter players for awhile until it is seen as another layer.

**Comment by CPTANT** [comment]
The windoors in Meta's EVA also stop nobody from just smashing them.

**Pull request merged**